internal/gps/version.go

```diff
@@ -274,6 +282,14 @@ func (v branchVersion) Pair(r Revision) PairedVersion {
 	}
 }
 
+func (v branchVersion) equals(c Constraint) bool {
+	v2, ok := c.(branchVersion)
+	if !ok {
+		return false
+	}
+	return v == v2
```

Comment:
> the default branch property on branchVersion is only set correctly by sourceMgr, not by anything that parses manifests.
The persistent cache handles this case (and hopefully all the other edge cases!).